specification/security/resource-manager/Microsoft.Security/preview/2020-01-01-preview/connectors.json

```diff
@@ -0,0 +1,618 @@
+      }
+    },
+    "HybridComputeSettingsProperties": {
+      "type": "object",
+      "description": "Settings for hybrid compute management",
+      "properties": {
+        "hybridComputeProvisioningState": {
+          "description": "State of the service principal and its secret",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Valid",
+            "Invalid",
+            "Expired"
+          ],
+          "x-ms-enum": {
+            "name": "hybridComputeProvisioningState",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Valid",
+                "description": "Valid service principal details."
+              },
+              {
+                "value": "Invalid",
+                "description": "Invalid service principal details."
+              },
+              {
+                "value": "Expired",
+                "description": "the service principal details are expired"
+              }
+            ]
+          }
+        },
+        "autoProvision": {
+          "type": "string",
+          "description": "Whether or not to automatically install Azure Arc (hybrid compute) agents on machines",
+          "enum": [
+            "On",
+            "Off"
+          ],
+          "x-ms-enum": {
+            "name": "autoProvision",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "On",
+                "description": "Install missing Azure Arc agent on VMs automatically"
+              },
+              {
+                "value": "Off",
+                "description": "Do not install Azure Arc agent on the VMs automatically"
+              }
+            ]
+          }
+        },
+        "resourceGroupName": {
+          "type": "string",
+          "description": "The name of the resource group."
+        },
+        "region": {
+          "type": "string",
+          "description": "Resource region",
+          "x-ms-mutability": [
+            "create",
+            "read"
+          ]
+        },
+        "proxyServer": {
+          "type": "object",
+          "description": "For a non-Azure machine that is not connected directly to the internet, specify a proxy server that the non-Azure machine can use.",
+          "$ref": "#/definitions/ProxyServerProperties"
+        },
+        "servicePrincipal": {
+          "description": "An object to access resources that are secured by an Azure AD tenant.",
+          "type": "object",
+          "$ref": "#/definitions/ServicePrincipalProperties"
+        }
+      },
+      "required": [
+        "autoProvision"
+      ]
+    },
+    "ServicePrincipalProperties": {
+      "type": "object",
+      "description": "Details of the service principal.",
+      "properties": {
+        "applicationId": {
+          "type": "string",
+          "description": "Application id of service principal."
+        },
+        "secret": {
```

Comment:
please ensure this value or any other secret value is not returned int he response of GET/PUT

Reply:
No secrets are returned, in the examples, these secrets are empty strings